### PR TITLE
feat:  Upload other applications (FPLA-2744)(FPLA-2808)

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-2744
https://tools.hmcts.net/jira/browse/FPLA-2808

### Change description ###
Upload other applications in `upload additional applications` event. 

- Removed feature toggle for **`upload additional applications`** which is added in #2362 (UploadC2DocumentsService.java and the UploadC2DocumentsServiceTest classes are reverted to master branch version.`UploadAdditionalApplicationsService` class is used for the new event, so feature toggle is no longer needed.)
- Removed PBA fields from CaseData. PBA fields are moved to `PBAPayment.java`.
- Fee mock responses are added in https://github.com/hmcts/fpla-docker/pull/59

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-definition/AuthorisationComplexType/caseDocuments/C2DocumentBundle.json [hunk 1]
```diff
@@ -391,65 +391,5 @@
...
-      {
-        "UserRoles": [
-          "caseworker-publiclaw-gatekeeper",
-          "caseworker-publiclaw-judiciary",
-          "caseworker-publiclaw-magistrate",
-          "[BARRISTER]"
-        ],
-        "CRUD": "R"
-      }
-    ]
```
Comment (aliveni) [inline ccd-definition/AuthorisationComplexType/caseDocuments/C2DocumentBundle.json:453]: these fields are new and should exist only in *-nonprod file

File: ccd-definition/CaseEventToFields/CareSupervision/uploadAdditionalApplications-nonprod.json [hunk 2]
```diff
@@ -94,52 +108,12 @@
...
-    "PageFieldDisplayOrder": 6,
-    "DisplayContext": "OPTIONAL",
-    "PageID": 3,
-    "PageDisplayOrder": 3,
+    "DisplayContext": "COMPLEX",
+    "PageID": 4,
+    "PageDisplayOrder": 4,
     "PageColumnNumber": 1,
-    "ShowSummaryChangeOption": "Y",
-    "FieldShowCondition": "usePbaPayment=\"Yes\""
```
Comment (aliveni) [inline ccd-definition/CaseEventToFields/CareSupervision/uploadAdditionalApplications-nonprod.json:143]: moved to `ccd-definition/CaseEventToComplexTypes/upload-c2/PBAPayment-nonprod.json`

File: ccd-definition/CaseTypeTab/CareSupervision/11_C2-nonprod.json [hunk 3]
```diff
@@ -17,7 +17,17 @@
...
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "Channel": "CaseWorker",
+    "TabID": "OtherApplicationsTab",
+    "TabLabel": "Other applications",
+    "TabDisplayOrder": 11,
+    "CaseFieldID": "c2DocumentBundle",
+    "TabFieldDisplayOrder": 3
```
Comment (aliveni) [inline ccd-definition/CaseTypeTab/CareSupervision/11_C2-nonprod.json:31]: existing cases with old c2 data should appear at the bottom of the tab (discussed with UX)

File: ccd-definition/AuthorisationCaseField/CareSupervision/gatekeeper.json [hunk 4]
```diff
@@ -2287,5 +2287,40 @@
     "CaseFieldID": "draftOrder10Document",
     "UserRole": "caseworker-publiclaw-gatekeeper",
     "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/01/2017",
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseField/CareSupervision/gatekeeper.json:2292]: most of these fields seems to be temporary fields that are not saved in case data. If gatekeeper cannot start event then there is no need to grant access to these fields

Comment (aliveni) [reply ccd-definition/AuthorisationCaseField/CareSupervision/gatekeeper.json:2292]: Removed temporary fields authorisation entries for gatekeeper, magistrate, judiciary and barristor roles

File: e2e/tests/hmctsAdministersCaseAfterSubmission_@slow_test.js [hunk 5]
```diff
@@ -57,51 +60,78 @@ Scenario('HMCTS admin amends children, respondents, others, international elemen
...
+  I.seeInTab(['Additional applications 1', 'Other applications', 'Supporting documents 1', 'File'], 'mockFile.txt');
+
+  I.seeInTab(['Additional applications 1', 'PBA Payment', 'Paid with PBA'], 'Yes');
+  I.seeInTab(['Additional applications 1', 'PBA Payment', 'Payment by account (PBA) number'], c2Payment.pbaNumber);
+  I.seeInTab(['Additional applications 1', 'PBA Payment', 'Client code'], c2Payment.clientCode);
+  I.seeInTab(['Additional applications 1', 'PBA Payment', 'Customer reference'], c2Payment.customerReference);
 });
 
-Scenario('HMCTS admin edits supporting evidence document on C2 application', async({I, caseViewPage, manageDocumentsEventPage}) => {
+xScenario('HMCTS admin edits supporting evidence document on C2 application', async({I, caseViewPage, manageDocumentsEventPage}) => {
```
Comment (aliveni) [inline e2e/tests/hmctsAdministersCaseAfterSubmission_@slow_test.js:134]: Manage documents populates c2 documents from `c2DocumentBundle`. Populating c2 data from the new case field is out of scope for [FPLA-2744](https://tools.hmcts.net/jira/browse/FPLA-2744). It will be implemented in [FPLA-2763](https://tools.hmcts.net/jira/browse/FPLA-2763)

File: ccd-definition/AuthorisationCaseField/CareSupervision/barrister.json [hunk 6]
```diff
@@ -357,5 +357,12 @@
     "CaseFieldID": "correspondenceDocumentsLANC",
     "UserRole": "[BARRISTER]",
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "additionalApplicationsBundle",
+    "UserRole": "[BARRISTER]",
+    "CRUD": "RU"
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseField/CareSupervision/barrister.json:366]: should not it be just R?

File: ccd-definition/AuthorisationCaseField/CareSupervision/gatekeeper.json [hunk 7]
```diff
@@ -2287,5 +2287,12 @@
...
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "additionalApplicationsBundle",
+    "UserRole": "caseworker-publiclaw-gatekeeper",
+    "CRUD": "RU"
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseField/CareSupervision/gatekeeper.json:2296]: is it not only R?

File: ccd-definition/AuthorisationCaseField/CareSupervision/judiciary.json [hunk 8]
```diff
@@ -2287,5 +2287,12 @@
     "CaseFieldID": "draftOrder10Document",
     "UserRole": "caseworker-publiclaw-judiciary",
     "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "additionalApplicationsBundle",
+    "UserRole": "caseworker-publiclaw-judiciary",
+    "CRUD": "RU"
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseField/CareSupervision/judiciary.json:2296]: same

File: ccd-definition/AuthorisationCaseField/CareSupervision/magistrate.json [hunk 9]
```diff
@@ -1566,5 +1566,12 @@
     "CaseFieldID": "draftOrder10Document",
     "UserRole": "caseworker-publiclaw-magistrate",
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseFieldID": "additionalApplicationsBundle",
+    "UserRole": "caseworker-publiclaw-magistrate",
+    "CRUD": "RU"
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseField/CareSupervision/magistrate.json:1575]: R

File: ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json [hunk 10]
```diff
@@ -265,6 +265,13 @@
     "CaseFieldID": "correspondenceDocumentsLANC",
     "UserRole": "[SOLICITOR]",
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/01/2017",
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json:270]: from first file it looks that [SOLICITOR] should be able to create event, so not sure why case filed access is different that for other users that can create event

File: ccd-definition/AuthorisationComplexType/caseDocuments/AdditionalApplicationsBundle.json [hunk 11]
```diff
@@ -0,0 +1,825 @@
...
+        "CRUD": "R"
+      },
+      {
+        "UserRoles": [
+          "caseworker-publiclaw-gatekeeper",
+          "caseworker-publiclaw-judiciary",
+          "caseworker-publiclaw-magistrate",
+          "[BARRISTER]"
+        ],
+        "CRUD": "U"
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationComplexType/caseDocuments/AdditionalApplicationsBundle.json:391]: is that correct? can caseworker-publiclaw-gatekeeper update it?

File: ccd-definition/ComplexTypes/CareSupervision/1_SupportingDocuments/SupplementsBundle.json [hunk 12]
```diff
@@ -0,0 +1,52 @@
+[
```
Comment (tomaszpowroznik) [inline ccd-definition/ComplexTypes/CareSupervision/1_SupportingDocuments/SupplementsBundle.json:1]: it should be rather Supplement and not SupplementBundle, it keep only one document and not bundle. It would make consistent with java code as well

File: ccd-definition/ComplexTypes/CareSupervision/2_C2DocumentBundle-nonprod.json [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "C2DocumentBundle",
+    "ListElementCode": "description",
+    "FieldType": "TextArea",
+    "ElementLabel": "Notes",
+    "SecurityClassification": "Public"
+  },
+  {
```
Comment (tomaszpowroznik) [inline ccd-definition/ComplexTypes/CareSupervision/2_C2DocumentBundle-nonprod.json:43]: is this field needed at all?

File: ccd-definition/ComplexTypes/CareSupervision/AdditionalApplicationsBundle.json [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+    "Searchable": "N"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "AdditionalApplicationsBundle",
+    "ListElementCode": "uploadedDateTime",
+    "FieldType": "Text",
+    "ElementLabel": "Date and time of upload",
+    "SecurityClassification": "Public",
+    "FieldShowCondition": "c2DocumentBundle=\"Do Not Show\"",
```
Comment (tomaszpowroznik) [inline ccd-definition/ComplexTypes/CareSupervision/AdditionalApplicationsBundle.json:36]: there is convention that it is all capital 'DO NOT SHOW'

File: ccd-definition/CaseField/CareSupervision/caseField.json [hunk 15]
```diff
@@ -1525,5 +1553,80 @@
...
+    "LiveFrom": "01/01/2018",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "ID": "additionalApplicationsLabel",
+    "Label": "Use this feature to upload applications for additional orders. You might need to upload additional forms, depending on your application type.",
+    "FieldType": "Label",
+    "SecurityClassification": "Public",
+    "Searchable": "N"
+  },
+  {
+    "LiveFrom": "01/01/2018",
```
Comment (tomaszpowroznik) [inline ccd-definition/CaseField/CareSupervision/caseField.json:1586]: are these PBA related fields are needed? there is complex type that keeps this information. I dont see access control configured for these fields so assume not used

File: ccd-definition/FixedLists/CareSupervision/Supplements.json [hunk 16]
```diff
@@ -0,0 +1,44 @@
+[
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "Supplements",
```
Comment (tomaszpowroznik) [inline ccd-definition/FixedLists/CareSupervision/Supplements.json:4]: can it be in sync with java class, here it is Supplements, in java it is SupplementType. I would name it SupplementType

File: ccd-definition/AuthorisationCaseField/CareSupervision/barrister.json (hunk 6 above)
Comment (aliveni) [reply ccd-definition/AuthorisationCaseField/CareSupervision/barrister.json:366]: I applied the same access permissions as existing `c2DocumentBundle`

Comment (aliveni) [reply ccd-definition/AuthorisationCaseField/CareSupervision/barrister.json:366]: `c2DocumentBundle.supportingEvidenceDocument` field has access permissions as `U` in See the existing c2DocumentBundle authorisation complex fields [here](https://github.com/hmcts/fpl-ccd-configuration/blob/master/ccd-definition/AuthorisationComplexType/caseDocuments/C2DocumentBundle.json#L313)

File: ccd-definition/AuthorisationCaseField/CareSupervision/gatekeeper.json (hunk 7 above)
Comment (aliveni) [reply ccd-definition/AuthorisationCaseField/CareSupervision/gatekeeper.json:2296]: same as above. supportingDocumentsBundle field has `U` permissions so, the parent element should have `U` as well?

File: ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json (hunk 10 above)
Comment (aliveni) [reply ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json:270]: supportingEvidence Documents can be updated in manage documents event so they will need `U` access

Comment (tomaszpowroznik) [reply ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json:270]: question is why [SOLICITOR] does not have access to 
  {
    "LiveFrom": "01/01/2017",
    "CaseTypeID": "CARE_SUPERVISION_EPO",
    "CaseFieldID": "c2Type",
    "UserRole": "[LASOLICITOR]",
    "CRUD": "CRU"
  },
  like [LASOLICITOR] if both have right to create event

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+
+    @PostMapping("/validate/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleValidateMidEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        List<String> errors = new ArrayList<>();
+
+        if (caseData.getTemporaryPbaPayment() != null) {
+            var updatedPbaNumber = pbaNumberService.update(caseData.getTemporaryPbaPayment().getPbaNumber());
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:74]: maybe pbaNumberService should accept PBAPayment and returns updated PBAPayment

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+
+        additionalApplications.add(0, element(uploadC2DocumentsService.buildAdditionalApplicationsBundle(
+            caseData, c2DocumentBundle, otherApplicationsBundle)));
+
+        caseDetails.getData().put("additionalApplicationsBundle", additionalApplications);
+
+        removeTemporaryFields(caseDetails, TEMPORARY_C2_DOCUMENT,
+            "c2Type",
+            "additionalApplicationType",
+            "usePbaPayment",
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:112]: is there a filed like usePbaPayment set somewhere ?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+
+        if (hasC2Order(caseData)) {
+            c2DocumentBundle = uploadC2DocumentsService.buildC2DocumentBundle(caseData);
+        }
+
+        if (hasOtherApplicationsOrder(caseData)) {
+            otherApplicationsBundle = uploadC2DocumentsService.buildOtherApplicationsBundle(caseData);
+        }
+
+        additionalApplications.add(0, element(uploadC2DocumentsService.buildAdditionalApplicationsBundle(
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:104]: uploadC2DocumentsService.buildAdditionalApplicationsBundle could accept just case data and do internally check if there is c2DocumentBundle or otherApplicationsBundle

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java [hunk 20]
```diff
@@ -0,0 +1,179 @@
...
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/submitted")
+    public void handleSubmittedEvent(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        final PBAPayment pbaPayment = caseData.getAdditionalApplicationsBundle().get(0).getValue().getPbaPayment();
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:129]: caseData.getAdditionalApplicationsBundle().get(0) is repeated twice, can be

final AdditionalApplicationsBundle lastBundle = caseData.getAdditionalApplicationsBundle().get(0).getValue();

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java [hunk 21]
```diff
@@ -0,0 +1,179 @@
...
+
+        if (hasC2Order(caseData)) {
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:131]: does it mean that payment is not taken when there is only other application?

File: ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json (hunk 10 above)
Comment (aliveni) [reply ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json:270]: oh, got it now.. I'll add the missing authorisation entries

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsServiceTest.java [hunk 22]
```diff
@@ -0,0 +1,259 @@
...
+            otherSupplementsBundle, otherSupportingDocument);
+
+        PBAPayment pbaPayment = buildPBAPayment();
+
+        AdditionalApplicationsBundle actual = service.buildAdditionalApplicationsBundle(
+            CaseData.builder().temporaryPbaPayment(pbaPayment).build(),
+            c2DocumentBundle,
+            otherApplicationsBundle);
+
+        assertThat(actual).extracting("author", "c2DocumentBundle", "otherApplicationsBundle", "pbaPayment")
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsServiceTest.java:132]: maybe
        AdditionalApplicationsBundle expectedBundle = AdditionalApplicationsBundle.builder()
            .author(HMCTS)
            .uploadedDateTime(formatLocalDateTimeBaseUsingFormat(time.now(), DATE_TIME))
            .c2DocumentBundle(c2DocumentBundle)
            .otherApplicationsBundle(otherApplicationsBundle)
            .pbaPayment(pbaPayment)
            .build();

        assertThat(actual).isEqualTo(expectedBundle);

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsServiceTest.java [hunk 23]
```diff
@@ -0,0 +1,259 @@
...
+        assertThat(documentBundle.getDocument().getFilename()).isEqualTo(DOCUMENT.getFilename());
+        assertThat(documentBundle.getApplicationType()).isEqualTo(C1_PARENTAL_RESPONSIBILITY);
+        assertThat(documentBundle.getParentalResponsibilityType()).isEqualTo(PR_BY_FATHER);
+        assertThat(documentBundle.getAuthor()).isEqualTo(HMCTS);
+        assertThat(documentBundle.getSupportingEvidenceBundle()).hasSize(1);
+        assertThat(documentBundle.getSupplementsBundle()).hasSize(1);
+    }
+
+    private void assertSupplementsBundle(Supplement actual, Supplement expected) {
+        assertThat(actual)
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsServiceTest.java:186]: maybe
 assertThat(actual).isEqualTo(expected.toBuilder()
            .dateTimeUploaded(time.now())
            .uploadedBy(HMCTS)
            .build());

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/FeeServiceTest.java [hunk 24]
```diff
@@ -236,6 +258,141 @@ void resetInvocations() {
...
+
+            assertThat(feesData.getTotalAmount()).isEqualTo(BigDecimal.valueOf(35));
+            assertThat(getFirstFeeCode(feesData)).isEqualTo(PARENTAL_RESPONSIBILITY_FATHER);
+        }
+
+        private String getFirstFeeCode(FeesData feesData) {
+            return feesData.getFees().get(0).getCode();
+        }
+
+        @AfterEach
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/FeeServiceTest.java:390]: should not be needed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java (hunk 21 above)
Comment (SooryaSu) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:112]: If the user only selects other application then no event will published which means an email notification won't be sent out

File: ccd-definition/AuthorisationComplexType/caseDocuments/AdditionalApplicationsBundle.json (hunk 11 above)
Comment (aliveni) [reply ccd-definition/AuthorisationComplexType/caseDocuments/AdditionalApplicationsBundle.json:391]: gatekeeper can update the supporting evidence documents, so the parent field should have `U`

File: ccd-definition/ComplexTypes/CareSupervision/1_SupportingDocuments/SupplementsBundle.json (hunk 12 above)
Comment (aliveni) [reply ccd-definition/ComplexTypes/CareSupervision/1_SupportingDocuments/SupplementsBundle.json]: Renamed the complex type

File: ccd-definition/ComplexTypes/CareSupervision/AdditionalApplicationsBundle.json (hunk 14 above)
Comment (aliveni) [reply ccd-definition/ComplexTypes/CareSupervision/AdditionalApplicationsBundle.json]: updtaed

File: ccd-definition/CaseField/CareSupervision/caseField.json (hunk 15 above)
Comment (aliveni) [reply ccd-definition/CaseField/CareSupervision/caseField.json:1577]: Removed pba fields

File: ccd-definition/FixedLists/CareSupervision/Supplements.json (hunk 16 above)
Comment (aliveni) [reply ccd-definition/FixedLists/CareSupervision/Supplements.json]: renamed it as SupplementType

File: service/src/main/java/uk/gov/hmcts/reform/fnp/model/fee/FeeType.java [hunk 25]
```diff
@@ -38,6 +69,42 @@
         );
     }
 
+    private static final Map<SupplementType, FeeType> supplementToFeeMap;
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fnp/model/fee/FeeType.java]: updated

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java (hunk 17 above)
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java]: added method to take PBAPayment as a parameter. I left the existing method as it is still used in uploadC2 event

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java (hunk 18 above)
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java]: no, it's not. cleaned up the temporary fields list

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java (hunk 19 above)
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java]: updated

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java (hunk 21 above)
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:115]: I've updated the submitted callback to take payment for C2/Other/both

File: service/src/main/java/uk/gov/hmcts/reform/fpl/enums/OtherApplicationType.java [hunk 26]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.fpl.enums;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@Getter
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/enums/OtherApplicationType.java]: removed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/common/AdditionalApplicationsBundle.java [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.jackson.Jacksonized;
+import uk.gov.hmcts.reform.fpl.model.PBAPayment;
+
+@Data
+@Builder(toBuilder = true)
+@Jacksonized
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/common/AdditionalApplicationsBundle.java]: removed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/common/OtherApplicationsBundle.java [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+
+    @Override
+    public List<Element<SupportingEvidenceBundle>> getSupportingEvidenceNC() {
+        return getSupportingEvidenceBundle().stream()
+            .filter(doc -> !doc.getValue().isConfidentialDocument())
+            .collect(Collectors.toList());
+    }
+
+    @JsonIgnore
+    public String getAllDocumentFileNames() {
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/common/OtherApplicationsBundle.java]: removed all unused methods

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/ApplicationsFeeCalculator.java [hunk 29]
```diff
@@ -0,0 +1,104 @@
...
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ApplicationsFeeCalculator {
+
+    private static final String DISPLAY_AMOUNT_TO_PAY = "displayAmountToPay";
+    private static final String AMOUNT_TO_PAY = "amountToPay";
+
+    private final FeeService feeService;
+
+    public Map<String, Object> calculateFee(CaseData caseData) {
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/ApplicationsFeeCalculator.java:43]: done

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/ApplicationsFeeCalculator.java [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+            data.put(DISPLAY_AMOUNT_TO_PAY, NO.getValue());
+        }
+        return data;
+    }
+
+    private List<Element<Supplement>> mergeSupplementsBundles(CaseData caseData) {
+        List<Element<Supplement>> supplementsBundle = new ArrayList<>();
+
+        if (caseData.getTemporaryC2Document() != null
+            && isNotEmpty(caseData.getTemporaryC2Document().getSupplementsBundle())) {
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/ApplicationsFeeCalculator.java]: `getSupplementsBundle()` is no longer null-safe as it was causing issues in the pro-version c2 bundle. so, I'll leave it as it is now. is it ok?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/payment/FeeService.java [hunk 31]
```diff
@@ -67,6 +85,74 @@ public FeesData getFeesDataForC2(C2ApplicationType c2ApplicationType) {
             .build();
     }
 
+    public FeesData getFeesDataForAdditionalApplications(
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/payment/FeeService.java:88]: I'll move it to ApplciationFeeCalculator

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/FeeServiceTest.java (hunk 24 above)
Comment (aliveni) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/FeeServiceTest.java:390]: Removed

File: ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json (hunk 10 above)
Comment (aliveni) [reply ccd-definition/AuthorisationCaseField/CareSupervision/solicitor.json:270]: Added the missing entries for Solicitor role

File: e2e/config.js [hunk 32]
```diff
@@ -156,6 +156,7 @@ module.exports = {
     closeTheCase: 'Close the case',
     returnApplication: 'Return application',
     manageDocuments: 'Manage documents',
+    uploadAdditionalApplications: 'Upload additional applications',
```
[the remark this answers is not in the document]
Comment (aliveni) [reply e2e/config.js:159]: It is still used in manage documents e2e tests

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/ApplicationsFeeCalculator.java (hunk 30 above)
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/ApplicationsFeeCalculator.java]: I've updated the code as below:
`ofNullable(c2Bundle.getSupplementsBundle()).ifPresent(
                bundle -> feeTypes.addAll(getSupplementsFeeTypes(bundle)));`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsService.java [hunk 33]
```diff
@@ -0,0 +1,109 @@
...
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class UploadAdditionalApplicationsService {
+
+    private final Time time;
+    private final DocumentUploadHelper documentUploadHelper;
+
+    public C2DocumentBundle buildC2DocumentBundle(CaseData caseData) {
```
[the remark this answers is not in the document]
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/additionalapplications/UploadAdditionalApplicationsService.java]: Updated as suggested

File: ccd-definition/ComplexTypes/CareSupervision/2_C2DocumentBundle-nonprod.json (hunk 13 above)
Comment (aliveni) [reply ccd-definition/ComplexTypes/CareSupervision/2_C2DocumentBundle-nonprod.json]: Removed all unused fields - description, representative name, pba fields in c2DocumentBundle

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java [hunk 34]
```diff
@@ -126,37 +108,39 @@ public void handleSubmittedEvent(
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
         CaseData caseData = getCaseData(caseDetails);
 
-        final PBAPayment pbaPayment = caseData.getAdditionalApplicationsBundle().get(0).getValue().getPbaPayment();
+        final AdditionalApplicationsBundle lastBundle = caseData.getAdditionalApplicationsBundle().get(0).getValue();
+
+        final PBAPayment pbaPayment = lastBundle.getPbaPayment();
 
         if (hasC2Order(caseData)) {
             C2DocumentBundle c2DocumentBundle = caseData.getAdditionalApplicationsBundle()
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java:116]: lastBundle

**Review by tomaszpowroznik — DISMISSED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java (hunk 34 above)
Comment (aliveni) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java]: good catch, updated the code to use `lastBundle` variable

File: ccd-definition/ComplexTypes/CareSupervision/2_C2DocumentBundle-nonprod.json (hunk 13 above)
Comment (aliveni) [reply ccd-definition/ComplexTypes/CareSupervision/2_C2DocumentBundle-nonprod.json:43]: @tomaszpowroznik I have to add these fields back to `C2DocumentBundle` complex type and add entries in authorisation complex type, as we have to support the existing case data with C2 applications.

File: e2e/tests/manageDocuments_test.js [hunk 35]
```diff
@@ -98,7 +98,7 @@ Scenario('HMCTS Admin and LA upload confidential and non confidential correspond
   assertCorrespondence(I, 'local authority', 2, 'C2 supporting document', 'Supports the C2 application');
 });
 
-Scenario('HMCTS Admin and LA upload confidential C2 supporting documents', async ({I, caseViewPage, manageDocumentsEventPage, manageDocumentsLAEventPage, uploadC2DocumentsEventPage}) => {
+xScenario('HMCTS Admin and LA upload confidential C2 supporting documents', async ({I, caseViewPage, manageDocumentsEventPage, manageDocumentsLAEventPage, uploadC2DocumentsEventPage}) => {
```
Comment (aliveni) [inline e2e/tests/manageDocuments_test.js:101]: I'll fix the e2e test in the manage documents ticket [FPLA-2763](https://tools.hmcts.net/jira/browse/FPLA-2763)

**Review by nraghu32 — APPROVED** [review]
